python/dask_cudf/dask_cudf/io/tests/test_s3.py

```diff
@@ -11,12 +14,22 @@
 
 import dask_cudf
 
-moto = pytest.importorskip("moto", minversion="1.3.14")
+moto = pytest.importorskip("moto", minversion="3.1.6")
 boto3 = pytest.importorskip("boto3")
 requests = pytest.importorskip("requests")
 s3fs = pytest.importorskip("s3fs")
 
 
+@pytest.fixture(scope="session")
+def endpoint_port():
+    # Return a free port per worker session.
+    sock = socket.socket()
+    sock.bind(("127.0.0.1", 0))
+    port = sock.getsockname()[1]
+    sock.close()
```

Comment:
Wait, is this just to get a port number of an open port? If so, I was misunderstanding how this worked earlier. The context manager was probably failing because the socket was still held by `sock` when you needed to close it and give up that port number to the `moto` server. If that is correct, you aren't supposed to _hold_ the socket resource with a context manager / `yield` fixture. Please use this approach and ensure that `sock` is closed before returning the port number.